src/include/network/postgres/portal.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "common/managed_pointer.h"
+#include "network/postgres/postgres_defs.h"
+#include "network/postgres/statement.h"
+#include "planner/plannodes/abstract_plan_node.h"
+#include "type/transient_value.h"
+
+namespace terrier::network {
+
+/**
+ * Portal is a postgres concept (see the Extended Query documentation:
+ * https://www.postgresql.org/docs/current/protocol-flow.html#PROTOCOL-FLOW-EXT-QUERY)
+ * It encapsulates a reference to its originating statement, the parameters (if any), the output formats, and the
+ * optimized physical plan. It represents a query ready to be executed.
+ */
+class Portal {
```

Comment:
Is this just for Postgres? Maybe we can name it more specifically, something like postgres_portal?